src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +34,132 @@ const ActorInit Bg_Jya_Lift_InitVars = {
     (ActorFunc)BgJyaLift_Destroy,
     (ActorFunc)BgJyaLift_Update,
     (ActorFunc)BgJyaLift_Draw,
+
 };
 */
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899BA0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Init.s")
+    extern s16      D_8089A020;
+    extern s32      D_8089A044;
+    extern s32      D_0600D7E8;
+    extern UNK_TYPE D_0600CCE0;
+
+void BgJyaLiftInitHelper(BgJyaLift* this, GlobalContext* globalCtx, u32 arg2, DynaPolyMoveFlag moveFlag) { //seems to help set up the actor
+    s32 pad1; // padding that makes everything match nicely
+    u32 localConst;
+    localConst = 0U;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, (void*)&localConst);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, localConst);
+}
+
+void BgJyaLift_Init(Actor* thisx, GlobalContext* globalCtx) {
+     BgJyaLift* this = THIS;
+    THIS->unk16A = 0;
```

Comment:
is the usage of the macro directly needed here? id think you can just use this (not thisx) for the whole thing and be fine